### PR TITLE
DRAFT: CI/CD pipelines for Kubernetes apps and infrastructure with Codefresh and Pulumi

### PR DESCRIPTION
**Pull request by spara**

### CODE REVIEW

**Review by MitchellGerdisch — COMMENTED** [review]
Other than the to-be-filled <ADD CODEFRESH CONTENT HERE> bit, it looks good to me.

File: content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+Continuous integration is an automated process that begins with a code push to a repository. From there, the code can be built and tested and eventually released as an artifact. The CI process is determined by your organization and it can include manual testing of the code to catch corner cases.
+
+A pipeline is the deployable unit path for CI/CD. A pipeline starts when code is committed to a repository like GitHub. Next step is a notification to a build system, such as Codefresh. The build system compiles the code and runs unit tests. If your code passes the unit tests, integration tests are the next step. In the case of containers, if your code passes both unit and integration tests, the images will be created and pushed into a registry service. This is the simplest example of a pipeline, but you can do many more things such as security scans, check modules for CVEs (Common Vulnerabilities and Exploits), send Slack notifications, and run quality checks. A pipeline can be fully automated or have checkpoints that require approval before resuming.
+
+< ADD CODEFRESH CONTENT HERE >
```
Comment (praneetloke) [inline content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md:29]: Not sure if @kostis-codefresh was planning to add anything here.

```suggestion
```

Comment (kostis-codefresh) [reply content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md:29]: YesI have some content. @spara I have updated our google doc. Please see changes there.

Comment (spara) [reply content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md]: added

File: content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- [iam.py](https://github.com/pulumi/examples/blob/master/aws-py-eks/iam.py) creates roles for the cluster, and the nodegroup and policy attachments
+- [utils.py](https://github.com/pulumi/examples/blob/master/aws-py-eks/utils.py) creates a `kubeconfig` file that can be used with `kubectl` to manage Kubernetes
+
+The main program (__main__.py) creates the Kubernetes cluster by calling the modules to deploy a cluster and a nodegroup with the policies applied. Using modules, we can make changes to the vpc, networking, roles, and policies and customize it to our application. In a CI/CD, all these changes would be pushed through a pipeline before deployment.
+
+Although this example is written in Python for AWS, there are similar examples in other languages and cloud providers in [Pulumi Examples](https://github.com/pulumi/examples).
+
+## Putting it Together
+
+This article described how we could build both applications and infrastructure with CI/CD pipelines. To learn more about how CodeFresh and Pulumi can automate Kubernetes deployments join us for a [technical session](https://www.pulumi.com/resources/ci-cd-pipelines-for-kubernetes-apps-with-codefresh/) on January 21st.
```
Comment (praneetloke) [inline content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md:66]: George had a suggestion (from the email thread) about moving this to the top. Maybe it could just simply say `Join us for a [technical session](link) ...`.

**Review by praneetloke — APPROVED** [review]

**Comment by kostis-codefresh** [comment]
@spara can you please reverse the order of the two pictures in the Codefresh section? The pipeline should be shown first and then the environments dashboard should be shown second.

File: content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md (hunk 2 above)
Comment (spara) [reply content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md:70]: Added CTA to before the cut line
